nickkirbus/technologies_test
# add tests task

What in this PR: 
1. Docker/docker-compose
2. a few EP for check
3. models for task
4. tests
5. custom command for seed db

Assumptions that were made due to the lack of necessity :
1. Not good scallable arhitecture (for task unnecessary, I think)
2. Not set up swagger (not in task)
3. not pure requirements (need for future)
4. Not pure response (for task unnecessary, I think)
5. Not created docker-compose for test (Not need currently, because we not deploy this, for local check you can use local docker-container)

For start: 
1. clone project and checkout to test_task branch.
6. docker-compose up <-- container wiil run on PORT: 8002 you can edit this in docker. 
7. enter to the container
8. for seed db run in command shell(inside container):  python manage.py seed_db 
9.  in container shell type  export DJANGO_SETTINGS_MODULE=pilot_project.settings
10. for run test in container shell type: pytest
11. for get list loan_applications: GET http://127.0.0.1:<your_port or 8002>/loan-applications/
12. for get unique manufacturer in loan_application: GET http://127.0.0.1:8002/loan-applications/contract/<contract_id>/
Also for checking of query you can:
1. In docker container shel run django shel: python manage.py shell
2. type: from django.db import connection
3. import model that you need in query
4. type: Product.objects.select_related('loan_application__contract', 'manufacturer').filter(loan_application__contract__id=contract_id).values('manufacturer').annotate(manufacturer_cnt=Count('id'))
5. type: connection.queries 
6. check
